### DISCUSSION TITLE
Error when trying to upload file to Supabase: `invalid input syntax for type uuid`

### DISCUSSION THREAD

**Discussion by magnusrodseth** [supabase/supabase#13534] [Questions] [answered]
I have a Next.js application using Supabase for data and file storage, and Clerk for user storage and authentication. Because users are stored with an ID like `user_abcdefghijkl`, and not a UUId like `123e4567-e89b-12d3-a456-426614174000`, I get the following error when trying to upload a file to Supabase:

```json
{
  "statusCode":"500",
  "error":"Database Error",
  "message":"invalid input syntax for type uuid: \"user_abcdefghijkl\""
}
```

This is the error that I `console.log` when calling the following:

```ts
  await supabase.storage.from(BUCKET_ID).upload(path, file, {
    cacheControl: "3600",
    contentType: "application/pdf",
    upsert: true,
  });
```

All other data can be successfully written to and read from the database. 

It seems to me like the call to `.upload(...)` implicitly checks the user ID of the currently logged-in user, and throws because the user ID is not a UUID. 

Note that the `userId` is stored as `TEXT` in the database, due to the way Clerk generated IDs.

Currently, during development, I allow all users to select, insert, update and delete. Hence, I don't think the error stems from a policy that's too restrictive.

**How can I solve this issue and allow users to upload files?**

**Comment by GaryAustin1** [comment, marked as answer]
The owner column in storage is type UUID and is a foreign key to auth.users id column(also uuid).   The jwt sub claim goes into that on a storage API call.

I see you found the related issue.
https://github.com/supabase/storage-api/issues/276

The user suggestion to change the storage.objects column type and drop the foreign key is risky as Supabase could wipe those changes, or not work with a non-UUID owner column at any point.

**Reply by magnusrodseth** [reply]
After some debugging after publishing this issue, I came to the same conclusion.
